packages/styled-components/src/models/GlobalStyle.js

```diff
@@ -18,28 +18,24 @@ export default class GlobalStyle {
     this.rules = rules;
     this.componentId = componentId;
     this.isStatic = isStaticRules(rules, EMPTY_ARRAY);
-
-    if (!StyleSheet.master.hasId(componentId)) {
-      StyleSheet.master.deferredInject(componentId, []);
-    }
+    StyleSheet.registerId(componentId);
   }
 
   createStyles(executionContext: Object, styleSheet: StyleSheet) {
     const flatCSS = flatten(this.rules, executionContext, styleSheet);
     const css = stringifyRules(flatCSS, '');
+    const id = this.componentId;
 
-    styleSheet.inject(this.componentId, css);
+    // NOTE: We use the id as a name as well, since these rules never change
```

Comment:
cGS can include function interpolations

Reply:
@probablyup Yes but in the way they're used they're "static"